--- a/Common/Core/vtkGenericDataArrayLookupHelper.h
+++ b.mod/Common/Core/vtkGenericDataArrayLookupHelper.h
@@ -25,6 +25,7 @@
 #include "vtkIdList.h"
 #include <algorithm>
 #include <cmath>
+#include <limits>
 #include <unordered_map>
 #include <vector>
 
--- a/Common/DataModel/vtkPiecewiseFunction.cxx
+++ b.mod/Common/DataModel/vtkPiecewiseFunction.cxx
@@ -22,6 +22,7 @@
 #include <cassert>
 #include <cmath>
 #include <iterator>
+#include <limits>
 #include <set>
 #include <vector>
 
--- a/Filters/HyperTree/vtkHyperTreeGridThreshold.cxx
+++ b.mod/Filters/HyperTree/vtkHyperTreeGridThreshold.cxx
@@ -28,6 +28,7 @@
 #include "vtkHyperTreeGridNonOrientedCursor.h"
 
 #include <cmath>
+#include <limits>
 
 vtkStandardNewMacro(vtkHyperTreeGridThreshold);
 
--- a/Rendering/Core/vtkColorTransferFunction.cxx
+++ b.mod/Rendering/Core/vtkColorTransferFunction.cxx
@@ -21,6 +21,7 @@
 #include <algorithm>
 #include <cmath>
 #include <iterator>
+#include <limits>
 #include <set>
 #include <vector>
